### PR TITLE
test: Add more test cases for HIP-1215

### PR DESCRIPTION
**Pull request by stoyanov-st**
**Description**:

Adding additional test cases for HIP-1215. 

**Related issue(s)**:

Fixes #

**Notes for reviewer**:
Real functionality checks, corner cases and deleteSchedule tests will be added once the CN implementation is done.

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: system-contract-testing/test/hip-1215/hip1215-test.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ describe("HIP-1215 System Contract testing", () => {
 
   describe("scheduleCall", () => {
     it("should schedule a call", async () => {
-      await mock1215.setResponse(true);
+      await mock1215.setResponse(true, 22);
       const tx = await hip1215.scheduleCallWithFullParam(
         "0x0000000000000000000000000000000000000167",
```
Comment (gkozyryatskyy) [inline system-contract-testing/test/hip-1215/hip1215-test.js:23]: move static addr to constant?

File: system-contract-testing/test/hip-1215/hip1215-test.js [hunk 2]
```diff
@@ -35,11 +35,1004 @@ describe("HIP-1215 System Contract testing", () => {
...
+        GAS_LIMIT_1_000_000.gasLimit,
+        0,
+        "0x5b8f8584"
+      );
+      const transaction = await tx.wait();
+      const log = transaction.logs.find(
+        e => e.fragment.name == Events.ScheduleCall
+      );
+      expect(log.args[0]).to.equal(22n);
+      expect(log.args[1]).to.equal("0x000000000000000000000000000000000000007B");
```
Comment (gkozyryatskyy) [inline system-contract-testing/test/hip-1215/hip1215-test.js:72]: move static addr to constant? same for others

**Review by gkozyryatskyy — APPROVED** [review]
LGTM

**Review by gkozyryatskyy — APPROVED** [review]

File: system-contract-testing/test/hip-1215/hip1215-test.js [hunk 3]
```diff
@@ -31,15 +33,1008 @@ describe("HIP-1215 System Contract testing", () => {
...
+      // With mocked setup we can only verify that the txn was success/revert
+      expect(transaction.status).to.equal(1);
+    });
+
+    it("should fail with gasLimit 1000", async () => {
+      await mock1215.setResponse(false, 30);
+      const tx = await hip1215.scheduleCallWithFullParam(
+        htsAddress,
+        new Date().getUTCMilliseconds() + 100,
+        100,
```
Comment (konstantinabl) [inline system-contract-testing/test/hip-1215/hip1215-test.js:122]: test says gasLimit 1000, but its set to 100, maybe rename the test or change the value. Also why not just add it to a constant and use it as done here https://github.com/hashgraph/hedera-evm-testing/pull/8/files#diff-3d915328410c3fbbdd0b7dbc36c69828b7f0590660d39afb0c5c58ab254b0718R84

File: system-contract-testing/test/hip-1215/hip1215-test.js [hunk 4]
```diff
@@ -31,15 +33,1008 @@ describe("HIP-1215 System Contract testing", () => {
...
+      );
+      expect(log.args[0]).to.equal(366n);
+      expect(log.args[1]).to.equal(ethers.ZeroAddress);
+      // With mocked setup we can only verify that the txn was success/revert
+      expect(transaction.status).to.equal(1);
+    });
+
+    it("should fail with amount more than contract balance", async () => {
+      await mock1215.setResponse(false, 10);
+      expect(await signers[0].provider.getBalance(await hip1215.getAddress())).to.eq(0);
```
Comment (konstantinabl) [inline system-contract-testing/test/hip-1215/hip1215-test.js:157]: await should be before expect, otherwise this could produce false positives

File: system-contract-testing/test/hip-1215/hip1215-test.js [hunk 5]
```diff
@@ -31,15 +33,1008 @@ describe("HIP-1215 System Contract testing", () => {
...
+      await signers[0].sendTransaction({
+        to:  await hip1215.getAddress(),
+        value: ethers.parseEther("5"),
+      });
+      const tx = await hip1215.scheduleCallWithFullParam(
+        htsAddress,
+        new Date().getUTCMilliseconds() + 100,
```
Comment (konstantinabl) [inline system-contract-testing/test/hip-1215/hip1215-test.js:164]: This is used on multiple occassions, can be moved to a constant with proper variable name

File: system-contract-testing/test/hip-1215/hip1215-test.js [hunk 6]
```diff
@@ -31,15 +33,1008 @@ describe("HIP-1215 System Contract testing", () => {
...
+    });
+
+    it("should succeed with amount sent to contract", async () => {
+      await mock1215.setResponse(true, 22);
+      const tx = await hip1215.scheduleCallWithFullParam(
+        htsAddress,
+        new Date().getUTCMilliseconds() + 100,
+        GAS_LIMIT_1_000_000.gasLimit,
+        ethers.parseEther('5'),
+        "0x5b8f8584",
```
Comment (konstantinabl) [inline system-contract-testing/test/hip-1215/hip1215-test.js:186]: This seems like a "magic value", can be moved to a constant

File: system-contract-testing/test/hip-1215/hip1215-test.js [hunk 7]
```diff
@@ -31,15 +33,1008 @@ describe("HIP-1215 System Contract testing", () => {
...
+      await new Promise(resolve => setTimeout(resolve, 100));
+      // needs CN for this expectation
+      // expect(await hip1215.getValue()).to.equal(63n);
+    });
+
+  });
+
+  describe("scheduleCallWithPayer()", () => {
+    it("should schedule a call with payer", async () => {
+      await mock1215.setResponse(true, 22);
```
Comment (konstantinabl) [inline system-contract-testing/test/hip-1215/hip1215-test.js:324]: The test cases expecting successful response - a.k.a response code 22, can be put in a separate before where this setResponse is set. This will remove redundant code and make code easier for maintenance

File: system-contract-testing/test/hip-1215/hip1215-test.js [hunk 8]
```diff
@@ -31,15 +33,1008 @@ describe("HIP-1215 System Contract testing", () => {
...
+        e => e.fragment.name == Events.ScheduleCall
+      );
+      expect(log.args[0]).to.equal(366n);
+      expect(log.args[1]).to.equal(ethers.ZeroAddress);
+      expect(transaction.status).to.equal(1);
+    });
+
+    it("should fail with amount more than contract balance", async () => {
+      await mock1215.setResponse(false, 10);
+      // expect(await signers[0].provider.getBalance(await hip1215.getAddress())).to.eq(0);
```
Comment (konstantinabl) [inline system-contract-testing/test/hip-1215/hip1215-test.js:477]: remove comment

**Review by konstantinabl — CHANGES_REQUESTED** [review]
Left a few comments
